### PR TITLE
449 - Fetcher service

### PR DESCRIPTION
**Pull request by avnersorek**
Work in progress for fetcher service

### CODE REVIEW

File: fetcher/src/handlers/reports-handler.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  return {
+    fromQuarter: queryString.fromQuarter,
+    toQuarter: queryString.toQuarter,
+    fromYear: queryString.year,
+    toYear: queryString.year,
+    statusReport: 1 // הוגש
+  };
+}
+
+export default handleReportsRequest;
```
Comment (RoySegall) [inline fetcher/src/handlers/reports-handler.ts]: let's add an empty line at the end of the file

File: fetcher/src/server.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import express from "express";
+import "express-async-errors";
+require("dotenv").config();
+
+import reportsHandler from "handlers/reports-handler";
+
+const app = express();
+const port = parseInt(process.env.PORT);
+
+app.get("/reports", reportsHandler);
```
Comment (RoySegall) [inline fetcher/src/server.ts]: We need a grpahql integration so we could query the service for paramerters: quarters, years, fund bodies etc.etc.

**Comment by RoySegall** [comment]
Looks good for now. Basically, after the process and kafaka merged, you should fire a kafka message to notify other services. The other services will listen to that event.

File: fetcher/src/server.ts (hunk 2 above)
Comment (avnersorek) [reply fetcher/src/server.ts]: Moved the entire thing to Graphql

File: fetcher/src/clients/s3-client.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+const Bucket = process.env.S3_BUCKET;
+const s3 = new AWS.S3();
+const ContentType = "application/vnd.ms-excel";
+
+export async function streamToS3(
+  row: ReportRow,
+  fileStream: ReadStream
+): Promise<string> {
+  if (!Bucket) throw new Error("Bucket must be defined");
```
Comment (RoySegall) [inline fetcher/src/clients/s3-client.ts]:
```suggestion
  if (!Bucket) {
    throw new Error("Bucket must be defined");
  }
```

File: fetcher/src/types/report-query.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  @Field(type => Int)
+  statusReport: number;
+
+  // TODO: not sure about these types
+  // corporation?: any;
+  // investmentName?: any;
+  // reportFromDate?: any;
+  // reportToDate?: any;
+  // reportType?: any;
+}
```
Comment (RoySegall) [inline fetcher/src/types/report-query.ts]:
```suggestion
}

```

**Comment by RoySegall** [comment]
Two things about where we going to store our files: 

- First, we have in Google something called "Cloud storage" which is the same as S3. 
- Second, we need to check if we in development mode or staging/production - when in development mode we need to keep the files on our local machine and serve it, for development and tests, and when we on production/staging environment we would keep the files on the cloud storage and serve them from there.

File: fetcher/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
+FROM node:11
```
Comment (RoySegall) [inline fetcher/Dockerfile]: Let's connect this one to the docker-compose

File: fetcher/src/types/report-query.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  @Field(type => Int)
+  fromYear: number;
+  @Field(type => Int)
+  toQuarter: number;
+  @Field(type => Int)
+  toYear: number;
+  @Field(type => Int)
+  statusReport: number;
+
+  // TODO: not sure about these types
```
Comment (RoySegall) [inline fetcher/src/types/report-query.ts]: If this not relevant - remove please.

File: fetcher/src/types/report-row.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  ReportHistLevel?: number;
+  ReportPeriodCode?: number;
+  ReportPeriodDesc?: string;
+  ShortName?: string;
+  StatusDate?: string;
+  SystemCode?: number;
+  SystemName?: string;
+  fileExt: string;
+  
+  // TODO: not sure about these types
```
Comment (RoySegall) [inline fetcher/src/types/report-row.ts]: If this not relevant - remove please.

Comment (avnersorek) [reply fetcher/src/types/report-row.ts]: done

File: fetcher/src/types/report-query.ts (hunk 6 above)
Comment (avnersorek) [reply fetcher/src/types/report-query.ts]: done

File: cms/src/web/profiles/open_pension/modules/open_pension_core/src/Controller/OpenPensionCoreAdminIndex.php [hunk 8]
```diff
@@ -38,6 +38,11 @@ public function index(): array {
         'title' => t('Reclamation tables'),
         'description' => t('Manage the reclamation tables'),
         'url' => Url::fromRoute('entity.instrument_type.collection'),
+      ],
+      [
+        'title' => t('Fetched links'),
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_core/src/Controller/OpenPensionCoreAdminIndex.php:43]: Move next to the uploaded files.

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/open_pension_fetcher.links.task.yml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+entity.open_pension_links.edit_form:
+  title: Edit
+  route_name: entity.open_pension_links.edit_form
+  base_route: entity.open_pension_links.canonical
+entity.open_pension_links.delete_form:
+  title: Delete
+  route_name: entity.open_pension_links.delete_form
+  base_route: entity.open_pension_links.canonical
+  weight: 10
+#entity.open_pension_links.collection:
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/open_pension_fetcher.links.task.yml:14]: Delete this block

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/open_pension_fetcher.routing.yml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+open_pension_fetcher.links:
+  path: '/api/fetcher-links'
+  defaults:
+    _title: 'Example'
+    _controller: '\Drupal\open_pension_fetcher\Controller\OpenPensionFetcherLinksController::build'
+  methods: ['PATCH', 'POST']
+  requirements:
+    _permission: 'access content'
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/open_pension_fetcher.routing.yml:8]: switch to a custom access requirements and make sure we can verify this came from our containers. Can be handled in another PR

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * @var Request|null
+   */
+  protected  $requestStack;
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php:27]: extra space.

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('request_stack')
+    );
+  }
+
+  public function getOrCreateFile(string $file_uri) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php:50]: add documentation

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+      return reset($file_ids);
+    }
+
+    $file = $this->entityTypeManager->getStorage('file')->create(['uri' => $file_uri]);
+    $file->save();
+
+    return $file->id();
+  }
+
+  public function getOrCreateMediaLink($file_id): \Drupal\Core\Entity\EntityInterface {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php:63]: Add documentation

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+        'field_media_file' => $file_id
+      ]);
+    $media->save();
+    return $media;
+  }
+
+  /**
+   * Builds the response.
+   */
+  public function build() {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php:85]: Rename

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+    // Get the payload.
+    $response = json_decode($this->requestStack->getContent(), TRUE);
+
+    if ($constraints = $this->validatePayload($this->requestStack->getMethod(), $response)) {
+      return $constraints;
+    }
+
+    if ($this->requestStack->getMethod() == Request::METHOD_POST) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php:93]: Add documentation

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+      return $this->createLink($response['link']);
+    }
+
+    // Create the file in the system.
+    return $this->setLinkToFile($response['link'], $response['file'], "{$response['name']}.XLSX");
+  }
+
+  public function validatePayload($method, $payload) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php:101]: Add docuentnation

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+    return null;
+  }
+
+  public function createLink($address) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php:105]: Add docuemtnation

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+      ->save();
+
+    if ($created) {
+      return new Response(t('File was updated'), Response::HTTP_CREATED);
+    }
+
+    return new Response(t('An error during creation'), Response::HTTP_BAD_REQUEST);
+  }
+
+  public function setLinkToFile($address, $file, $file_name) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php:125]: Add documantation

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+    if (!$link_ids = $this->getLinkEntityByAddress($address)) {
+      return new Response(t('File does not exits'), Response::HTTP_METHOD_NOT_ALLOWED);
+    }
+
+    $link = $this->entityTypeManager->getStorage('open_pension_links')->load(reset($link_ids));
+
+//    if ($link->get('open_pension_file')->value) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php:132]: Remove this

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php [hunk 20]
```diff
@@ -0,0 +1,160 @@
...
+
+    // Update the reference.
+    $link->set('open_pension_file', $media);
+    $link->save();
+
+    // update the object.
+    return new Response(t('File was updated'), Response::HTTP_OK);
+  }
+
+  public function getLinkEntityByAddress($address) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php:153]: Add doc and move to the other section of the API function

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php [hunk 21]
```diff
@@ -0,0 +1,188 @@
...
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity.repository'),
+      $container->get('entity_type.bundle.info'),
+      $container->get('datetime.time'),
+      $container->get('open_pension_fetcher.query')
+    );
+  }
+
+  protected function idLabelProcessor($items) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php:45]: add documentation

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php [hunk 22]
```diff
@@ -0,0 +1,188 @@
...
+      $processed_items = [];
+      foreach ($items as $item) {
+        $processed_items[$item['Id']] = $item['Label'];
+      }
+
+      return $processed_items;
+  }
+
+  protected function processYears($items) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php:54]: add dcocumentation

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php [hunk 23]
```diff
@@ -0,0 +1,188 @@
...
+    $processed_items = [];
+    foreach ($items as $item) {
+      $processed_items[$item] = $item;
+    }
+
+    return $processed_items;
+  }
+
+  protected function actions(array $form, FormStateInterface $form_state) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php:63]: `inhertidoc`

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php [hunk 24]
```diff
@@ -0,0 +1,188 @@
...
+    if ($this->getOperation() == 'edit') {
+      return $actions;
+    }
+
+    $actions['submit']['#submit'] = ['::submitForm', '::sendMutation'];
+
+    return $actions;
+  }
+
+  public function form(array $form, FormStateInterface $form_state) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php:75]: `inheritdoc`

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php [hunk 25]
```diff
@@ -0,0 +1,188 @@
...
+    // No need to display the URL.
+    $form['url']['#access'] = FALSE;
+    $form['open_pension_file']['#access'] = FALSE;
+
+    unset($form['created']);
+
+    return $form;
+  }
+
+  public function save(array $form, FormStateInterface $form_state) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php:146]: `inheritdoc`

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php [hunk 26]
```diff
@@ -0,0 +1,188 @@
...
+    }
+    else {
+      $this->messenger()->addStatus($this->t('The open pension links %label has been updated.', $message_arguments));
+      $this->logger('open_pension_fetcher')->notice('Updated new open pension links %label.', $logger_arguments);
+    }
+
+    $form_state->setRedirect('entity.open_pension_links.collection', ['open_pension_links' => $entity->id()]);
+  }
+
+  public function sendMutation(array $form, FormStateInterface $form_state) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Form/OpenPensionLinksForm.php:166]: Add doc

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/OpenPensionFetcherQuery.php [hunk 27]
```diff
@@ -0,0 +1,126 @@
...
+   * @param OpenPensionServicesAddresses $open_pension_health_status
+   *  The services addresses service.
+   */
+  public function __construct(ClientInterface $http_client, OpenPensionServicesAddresses $services_addresses, OpenPensionServicesHealthStatus $open_pension_health_status) {
+    $this->servicesAddresses = $services_addresses;
+    $this->httpClient = $http_client;
+    $this->servicesHealthStatus = $open_pension_health_status;
+  }
+
+  protected function sendQuery($query) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/OpenPensionFetcherQuery.php:37]: add doc

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/OpenPensionFetcherQuery.php [hunk 28]
```diff
@@ -0,0 +1,126 @@
...
+   *
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The HTTP client.
+   * @param OpenPensionServicesAddresses $open_pension_health_status
+   *  The services addresses service.
+   */
+  public function __construct(ClientInterface $http_client, OpenPensionServicesAddresses $services_addresses, OpenPensionServicesHealthStatus $open_pension_health_status) {
+    $this->servicesAddresses = $services_addresses;
+    $this->httpClient = $http_client;
+    $this->servicesHealthStatus = $open_pension_health_status;
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/OpenPensionFetcherQuery.php:34]: inject the messenger

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/OpenPensionFetcherQuery.php [hunk 29]
```diff
@@ -0,0 +1,126 @@
...
+      ],
+      'body' => json_encode([
+        'query' => $query,
+      ]),
+    ]);
+
+    return $response->getBody()->getContents();
+  }
+
+  public function query() {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/OpenPensionFetcherQuery.php:56]: add doc

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/OpenPensionFetcherQuery.php [hunk 30]
```diff
@@ -0,0 +1,126 @@
...
+        }
+      }
+    }
+    GRAPHQL;
+
+    $response = $this->sendQuery($query);
+    return json_decode($response, true)['data'];
+  }
+
+  public function mutate($system_field, $report_type, $from_year, $from_quarter, $to_year, $to_quarter) {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/OpenPensionFetcherQuery.php:93]: add do

File: cms/src/web/profiles/open_pension/modules/open_pension_services/src/OpenPensionServicesHealthStatus.php [hunk 31]
```diff
@@ -63,4 +63,22 @@ public function getProcessorState() {
       return self::SERVICE_IS_RESPONDING;
     }
   }
+
+  public function getFetcherState() {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_services/src/OpenPensionServicesHealthStatus.php:67]: add doc

File: fetcher/README.md [hunk 32]
```diff
@@ -0,0 +1,58 @@
...
+  ) {
+    links
+  }
+}
+``` 
+
+## Env file
+The env file should have the next values:
+
+    * foo - bar
```
Comment (RoySegall) [inline fetcher/README.md:58]: work on this

File: fetcher/src/clients/cma-api-client.ts [hunk 33]
```diff
@@ -0,0 +1,109 @@
...
+            "reportType": query.ReportType,
+            "systemField": query.SystemField,
+
+            "statusReport": 1,
+            "investmentName": null,
+            "reportFromDate": null,
+            "reportToDate": null
+        };
+
+        console.log(payload);
```
Comment (RoySegall) [inline fetcher/src/clients/cma-api-client.ts:39]: remove

File: fetcher/src/clients/kafka-client.ts [hunk 34]
```diff
@@ -0,0 +1,62 @@
+import kafka from "kafka-node";
+import { getKafkaHost, getKafkaTopic } from "services/config-service";
+
+export class KafkaClient {
```
Comment (RoySegall) [inline fetcher/src/clients/kafka-client.ts:4]: Might not needed for now

File: fetcher/src/services/cms-services.ts [hunk 35]
```diff
@@ -0,0 +1,40 @@
...
+                });
+
+                resolve();
+            }
+            catch (e) {
+                reject(e);
+            }
+        });
+    }
+}
```
Comment (RoySegall) [inline fetcher/src/services/cms-services.ts:40]: Add empty line

File: fetcher/tests/services/reports-service.test.ts [hunk 36]
```diff
@@ -0,0 +1,103 @@
...
+
+jest.mock("clients/kafka-client", () => ({
+    KafkaClient: mockKafkaClientCreate
+}));
+
+import {downloadReports} from "services/reports-service";
+import ReportQuery from "types/report-query";
+
+describe("Testing the reports service", () => {
+    // todo: check the CMS service was invoked.
```
Comment (RoySegall) [inline fetcher/tests/services/reports-service.test.ts:30]: do it

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/open_pension_fetcher.links.task.yml [hunk 37]
```diff
@@ -0,0 +1,14 @@
...
+entity.open_pension_links.edit_form:
+  title: Edit
+  route_name: entity.open_pension_links.edit_form
+  base_route: entity.open_pension_links.canonical
+entity.open_pension_links.delete_form:
+  title: Delete
+  route_name: entity.open_pension_links.delete_form
+  base_route: entity.open_pension_links.canonical
+  weight: 10
+
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/open_pension_fetcher.links.task.yml:14]: Remove extra line

File: fetcher/src/services/cms-services.ts [hunk 38]
```diff
@@ -0,0 +1,39 @@
...
+    constructor() {
+        this.api = axios.create({
+            baseURL: getCmsHost()
+        });
+    }
+    public async sendLinkAddress(link: string) {
+        try {
+            return await this.api.post('/api/fetcher-links',{link});
+        } catch (e) {
+            console.log(e);
```
Comment (RoySegall) [inline fetcher/src/services/cms-services.ts:18]: Remove console

**Comment by RoySegall** [comment]
So for now, with small amount of files we can send to file to the processor but when handling something like 1,000 files and above we get to some issues which one of them is that the server that holds the files locking us down.

the solution is that:
1. send the list of files to the CMS which can save huge amount of files
2. Download the first 50 files and send them
3. set a hook cron that will download each hour 50 files until we don't have any to download

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/composer.json [hunk 39]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/composer.json:1]: Remove

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php [hunk 40]
```diff
@@ -0,0 +1,278 @@
+<?php
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/Controller/OpenPensionFetcherLinksController.php:1]: Need to log the events

File: cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/OpenPensionFetcherQuery.php [hunk 41]
```diff
@@ -0,0 +1,222 @@
...
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\open_pension_services\OpenPensionServicesAddresses;
+use Drupal\open_pension_services\OpenPensionServicesHealthStatus;
+use GuzzleHttp\ClientInterface;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Defines a service provider for the Open Pension Fetcher module.
+ */
+class OpenPensionFetcherQuery {
```
Comment (RoySegall) [inline cms/src/web/profiles/open_pension/modules/open_pension_fetcher/src/OpenPensionFetcherQuery.php:17]: `OpenPensionFetcherQuery` => `OpenPensionFetcherService`

File: fetcher/src/services/reports-service.ts [hunk 42]
```diff
@@ -0,0 +1,81 @@
...
+/**
+ * Download files from a give links and send to the to the CMS.
+ *
+ * @param links
+ *  List of links.
+ */
+export function downloadLinks(links: any) {
+    links.map(async (item: any) => {
+        try {
+            console.log(`Downloading ${item['address']}`);
```
Comment (RoySegall) [inline fetcher/src/services/reports-service.ts:19]: Find a better way to log evnets

**Comment by RoySegall** [comment]
So for now, the fetcher with Drupal works awesome! there two things to cover:
* [ ] Make sure only request from the current container can send links
* [ ] The fetcher need to notify when the process has been done

**Pull request merged**